eclipse-theia/theia
# feat: add support for llamafile as ai model provider

#### What it does


This adds support to add llms run from llamafiles.

- The configured llamafiles are stored in the preferences
- The llamafile output is send to a new output channel <mymodel>-llamafile
- As a user you have to provide:
  - a custom modelname
  - the file uri to the llamafile
  - the port to run the llama.cpp server on

The current implementation does not support tools/functions so that agents like the workspace-agent don't work well with models provided via llamafiles.

#### How to test



Download a llamafile and follow the instructions on how to make it runnable from here: https://github.com/Mozilla-Ocho/llamafile?tab=readme-ov-file#quickstart
In Theia open the preferences, go to llamafile in the ai section and add a new entry.
You need to provide a name, uri (the file uri to the llamafile, e.g file:///home/me/Downloads/my.llamafile) and a port (e.g 30000).
Adopt one agent to use the new model, e.g the universal agent.
Then Call the "Start Llamafile" command.
See that there is a new output channel (my local model-llamafile).
Say "hi" in the chat and see your cpu/gpu working overtime to produce an answer ;-) .

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
